spec/controllers/work_packages/auto_completes_controller_spec.rb

```diff
@@ -203,38 +203,74 @@
         work_package_4
       end
 
-      context 'with scope all and cross project relations' do
-        let(:expected_values) { work_package_4 }
+      context 'with cross project relations' do
+        let(:project_id) { project.id }
 
         before do
           allow(Setting).to receive(:cross_project_work_package_relations?).and_return(true)
 
           get :index,
-              project_id: project.id,
+              project_id: project_id,
               q: work_package_4.id,
-              scope: 'all'
+              scope: scope
+        end
+
+        context 'with scope "relatable"' do
+          let(:scope) { 'relatable' }
+          let(:expected_values) { work_package_4 }
+
+          it_behaves_like 'successful response'
+
+          it_behaves_like 'contains expected values'
+
+          context 'without project_id' do
+            let(:project_id) { nil }
+
+            it 'returns HTTP Not Found' do
+              expect(response.status).to eql(404)
+            end
+          end
         end
 
-        it_behaves_like 'successful response'
+        context 'with scope "all"' do
+          let(:scope) { 'all' }
+          let(:expected_values) { work_package_4 }
```

Comment:
I'd have thought that with my suggestion you could implicitly do that exact check without much hassle.

Though if you really have to rewrite all the specs to make this possible, then this is probably something for another point in time.